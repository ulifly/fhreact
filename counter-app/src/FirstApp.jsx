
const nombre = ''
export const FirstApp = () => {

    return (
        <>
          <h1>{ nombre ? nombre : 'Sin nombre'  }</h1>
        </>
    );
}