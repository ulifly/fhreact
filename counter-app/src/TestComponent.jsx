import PropTypes from 'prop-types';
import React from 'react';

const TestComponent = ({ message }) => {
    return <h1>{message}</h1>;
}

TestComponent.propTypes = {
    message: PropTypes.string.isRequired
}

TestComponent.defaultProps = {
    message: 'Mensaje por defecto'
};

export default TestComponent;