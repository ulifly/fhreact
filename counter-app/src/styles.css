html, body {
  background-color: #21232a;
  color: white;
  font-family: Helvetica, Arial, sans-serif;
  font-size: 1.3rem;
  padding: 70px ;
}

button {
  background-color: #61dafb;
  border: none;
  border-radius: 5px;
  color: #282c34;
  cursor: pointer;
  font-size: 1.3rem;
  margin-top: 20px;
  padding: 10px 20px;
}