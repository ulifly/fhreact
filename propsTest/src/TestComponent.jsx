import PropTypes from 'prop-types';
import React from 'react';

export const TestComponent = ({ message }) => {
    return <h1>{message}</h1>;
}

TestComponent.propTypes = {
    message: PropTypes.string.isRequired
}

TestComponent.defaultProps = {
    message: 'Mensaje por defecto'
};

