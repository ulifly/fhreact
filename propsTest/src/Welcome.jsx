import React from "react";
import PropTypes from "prop-types";

export const Welcome = (props) => {
  console.log(props);
	return <h1>Hello, {props.name}</h1>;
};

Welcome.propTypes = {
  name: PropTypes.string.isRequired,
};

Welcome.defaultProps = {
  name: 'Stranger',
};
