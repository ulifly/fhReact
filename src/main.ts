import './style.css'

// import './bases/01-const-let'
// import './bases/02-template-string'
// import './bases/03-object-literal'
// import './bases/04-arrays'
// import './bases/05-functions'
// import './bases/06-obj-destructuring'
// import './bases/07-array-des'
// import './bases/08-import-export'
// import { getHeroesByOwner } from './bases/08-import-export'
// import { Owner } from './bases/data/heroes.data'
// import './bases/09-promises';
// import './bases/10-fetch-Api';
import './bases/11-async-await';


document.querySelector<HTMLDivElement>('#app')!.innerHTML = `
  <div>
    <h1>Hola mundo</h1>
  </div>
`

// console.log(getHeroesByOwner(Owner.DC));