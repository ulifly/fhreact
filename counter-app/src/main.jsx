
import React from 'react';
import ReactDOM from 'react-dom/client';
import PropTypes from 'prop-types';
import { FirstApp } from './FirstApp';
import './styles.css';

import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <CounterApp value= {0} /> */}
        <FirstApp />
    </React.StrictMode>
);