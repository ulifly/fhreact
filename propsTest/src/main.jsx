import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import React from 'react'

import {Welcome} from './Welcome'


createRoot(document.getElementById('root')).render(
  <StrictMode>
    <Welcome name = "Ulises"/>
    <Welcome />
  </StrictMode>,
)
