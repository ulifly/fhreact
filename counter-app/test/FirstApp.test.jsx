import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en el archivo FirstApp.test.jsx', () => {
    test('debe coincidir con el snapshot', () => {
        const nombre = 'Goku';
        const { container } = render(<FirstApp nombre="{nombre}"/>);

        expect(container).toMatchSnapshot();
        console.log()
        
    })
});
