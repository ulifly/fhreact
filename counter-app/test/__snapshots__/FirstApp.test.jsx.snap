// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo FirstApp.test.jsx debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    Hola 
    {nombre}
  </h1>
  <h2>
    No hay titulo
  </h2>
  <h3>
    No hay subtitulo
  </h3>
</div>
`;
